sklearn/tests/test_docstring_parameters.py

```diff
@@ -0,0 +1,160 @@
+    'sklearn.covariance',
+    # 'sklearn.cross_decomposition',
+    'sklearn.datasets',
+    'sklearn.decomposition',
+    # 'sklearn.ensemble',
+    'sklearn.feature_extraction',
+    # 'sklearn.feature_selection',
+    'sklearn.gaussian_process',
+    'sklearn.isotonic',
+    'sklearn.linear_model',
+    'sklearn.manifold',
+    'sklearn.multiclass',
+    'sklearn.metrics',
+    'sklearn.naive_bayes',
+    'sklearn.mixture',
+    # 'sklearn.model_selection',
+    'sklearn.neighbors',
+    'sklearn.neural_network',
+    'sklearn.preprocessing',
+    'sklearn.pipeline',
+    'sklearn.semi_supervised',
+    'sklearn.tree',
+    # 'sklearn.utils',
+]
+
+# functions to ignore args / docstring of
+_DOCSTRING_IGNORES = [
+    'sklearn.utils.deprecation.load_mlcomp',
+    'sklearn.pipeline.make_pipeline',
+    'sklearn.pipeline.make_union',
+    'sklearn.utils.extmath.safe_sparse_dot',
+    'RandomizedPCA',  # deprecated
+    'BaseForest',
+    'BaseDecisionTree',
+    'ExtraTreeClassifier',
+    'ExtraTreeRegressor',
+    'GaussianProcess',  # deprecated
+    'VBGMM',  # deprecated
+    'DPGMM',  # deprecated
+    'GMM',  # deprecated
+    'log_multivariate_normal_density',  # deprecated
+    'sample_gaussian',  # deprecated
+]
+
+_TAB_IGNORES = [
+]
+
+
+def test_docstring_parameters():
+    """Test module docstring formatting."""
+    try:
+        import numpydoc  # noqa
+    except ImportError:
+        raise SkipTest(
+            "numpydoc is required to test the docstrings")
+
+    from numpydoc import docscrape
+
+    incorrect = []
+    for name in PUBLIC_MODULES:
+        with warnings.catch_warnings(record=True):  # traits warnings
+            module = __import__(name, globals())
+        for submod in name.split('.')[1:]:
+            module = getattr(module, submod)
+        classes = inspect.getmembers(module, inspect.isclass)
+        for cname, cls in classes:
+            this_incorrect = []
+            if cname in _DOCSTRING_IGNORES:
+                continue
+            if cname.startswith('_') and cname not in _DOC_SPECIAL_MEMBERS:
+                continue
+            with warnings.catch_warnings(record=True) as w:
+                cdoc = docscrape.ClassDoc(cls)
+            if len(w):
+                raise RuntimeError('Error for __init__ of %s in %s:\n%s'
+                                   % (cls, name, w[0]))
+            if hasattr(cls, '__init__'):
+                this_incorrect += check_parameters_match(cls.__init__, cdoc)
+
+            for method_name in cdoc.methods:
+                method = getattr(cls, method_name)
+                param_ignore = None
+                # Now skip docstring test for y when y is None
+                # by default for API reason
+                # XXX remove transform
+                if method_name in ['fit', 'score', 'fit_predict',
+                                   'fit_transform', 'partial_fit',
+                                   'transform', 'inverse_transform',
```

Comment:
@raghavrv you need to remove now that the other PR is merged the transform and inverse_transform from here.

you should add to ignored PLS classes